### ISSUE TITLE
 Add option to Bury the 'Older' sibling ( the one which is being studied) when it is time to Unbury the Youngster.

### ISSUE THREAD

**Issue by dedu6ka** [lovac42/PushParadox#1]
Distinguish the terms 'Siblings'  and 'Related cards'. 
This is a perfect time to formulate the difference between these terms; they are in the Manual but not the definitions. 
     For me,  the siblings are cards to be learned independently from each other; the Related cards - to be learned one right after another.
    When i learn a Language, i stay with the Passive card for a few reps. Then i suspend it and unbury the sibling. The rational is -- if i now the Production card, i will most likely know the Recognition card; i don't want the Recognition card to clutter the deck; i move it to "Archive" deck.  
   Does it appeal to you ? Would you add this option?

**Comment by lovac42** [comment]
Siblings are cards that came from the same note. They have the same note id and can be tracked. Related cards are a figment of your imagination, but hard to proof. Connecting the dots between related cards is a lot harder then matching note ids.

Reverse-burying siblings, on the other hand is do-able, but gets technical with multi siblings involved. For now, you can just suspend the production card and the sibling will automatically be unburied and reviewed the next day.

In Anki, new cards data are reset once they graduate, so we could enter some pseudo data to force the reviewer to view the card. Say you suspend sibling A, it will trigger the addon and modifies (new card) sibling B's due count. With the due set to 0, the reviewer will pick card B as the next card to be reviewed. And card C, the other sibling, will be buried by this addon.

**Comment by dedu6ka** [comment]
>Related cards are a figment of your imagination ....
..   
FWIW, in case you did not see it, quote from the Manual:  
"these related cards are called siblings. "  
..  
To me they are the same; seeing both terms on your screenshot makes me stop and think ... 
So i suggested to define them in the way i did:  
* Siblings: do not learn them one after another, without spacing.  
* Related: do not space them apart; dont delay learning; do not bury Related cards.  
..  
> Reverse-burying siblings, on the other hand is do-able, but gets technical with 'multi' siblings involved.  
..  
My suggestion relates to Language-like cards only ( the Siblings - in my classification). So, there are only 2 cards.

**Comment by lovac42** [comment]
Anki seems to make an effort to distort the terminology (e.g. hard factor, ease bonus). If you're interested in wordings, take it up with DAE.

I believe what you are trying to describe would be similar to the "Swap QA" feature in supermemo?
```
Swap QA (Shift+Ctrl+S) - swap the question with the answer (as in standard flashcard programs). Usually you need this option to swap questions with answers when you learn bilingual pairs of vocabulary. It is most useful in conjunction with Duplicate element.
```
That is something I would like to get to as well, but need to think about how to design it on top of Anki.

**Comment by lovac42** [comment]
Under what condition would you suspend the passive card? Intervals or learning steps or something else? If you are to program a machine to automate this, what criteria would you use?

**Comment by lovac42** [comment]
paging dr. @dedu6ka https://github.com/lovac42/TheRoommateSwitch

**Comment by dedu6ka** [comment]
The following is from the top of my head. I would like to sleep on it for a day or two and submit some changes and screenshots and more details; and read your comments.  
My sugestion relates to Language cards ( can better handle Synonyms with this addon - by eliminating the interference).  
The Note type might have three FIelds positioned in the order of decreasing difficulties;  
...  
Fields are: 
- a naked word  (minus) exclusion answer:  to rander - perform -submit. These exclusions 
   shown here because they were used by the user during the first rep. 
- short phrase  
- Hint / Mnemonics  
...   
A card will be rated Good only after a correct answer for the Naked word.  
...  
I would suspend the first card type 'after' Three Good reps in Review status.  
The third rep will be a Scheduled rep; so actually we suspend the cars after two 'Confirned' reps.  
Would be nice to save the  third rep in the History table - to reflect it in the  Retention chart, the least.
And  
Have the option to suspend it at will -- if a card is too easy.  
..  
PS.  
To ensure the high level of retention for the Graduation interval,  
we have to have a good learning-steps-sequence, one that ensures 95-98% success rate for the Grad ivl.  
Can you track it ? Even a Table in the Options would do;  or  
Modify the Glutanimate's "True retention by interval" addon. or  
In Anki graph, add another chart "Graduatind ivl" ;  or  
Ib Anki graph, add chart "Step2 step3 Graduating ivl" chart would be the best. 
Of course these Learning stats are to be based on ACTUAL Delays not based on Reps.  
If you modify the "Pretzel" addon ( for "Today" stats )  and count the Cards - not Reps that would be great.  
As of now, the Today stats is a lie ( unless user clicks on 10min stap just once; but that is not the case for med students.  
..  
PPS.   
1. OVERDUE cards.  If i fail a card which was overdue >( bunch of 'if' statements ), 
.... will i be able to use one of 3 options from the "Rememorize" addon.

**Comment by dedu6ka** [comment]
This is the additional info to my previous post:  
Edit feb-27:  
If a user Reschedules a card, will you treat this as a Regular repetiton?  
FYI: Supermemo  keeps a lapsed card in relearning state for 3 reps.  
Regardless of what Supermemo says, i think the number of GOOD reps should be in the config file.  
DELETED this line:  
"Of course these Learning stats are to be based on ACTUAL Delays not based on Reps."  
End of edit.

**Comment by lovac42** [comment]
SM don't use learning steps. It's not even the correct SM terminology, so what were you saying?

Checking retention would not be possible for the intra-day time scale. We would need a different metric to measure those. That could take months or years just testing.

```
If a user Reschedules a card, will you treat this as a Regular repetiton?
```
That requires a grade from the user and not the intention of ReMemorize. It could easily be done with some sort of custom ivl addon, but I have no need for those as it distorts the algorithm.

**Try to rewrite it in a way that I can better understand you.** Keep irrelevant discussions out. Click on edit in the ... menu next to the smiley face.

**Comment by dedu6ka** [comment]
> SM don't use learning steps.   

I have said "Learning state" -  whatever it is in the SM. 
> Checking retention would not be possible for the intra-day time scale.  

I was not talking about the  "intra-day" steps. I am suggesting to track the number of Yes and NO answers  for the Graduating interval; that data would tell us whether or not we need to adjust the duration an the number of steps. Should be a walk in the park, right ?  
> Rescheduling reps  

Are you saying that the top 2 Resch reps ( see the table) will not be taken into account  when making a determination about Suspending+Unbury?

.........................      Date | T | R | Ivl.. | IntDate |........ Ease
2019-02-27@18:10 | Resch | 0 | 22d | 2019-03-21 | 240
2018-12-30@15:36 | Resch | 0 |.. 6d | 2019-01-05 | 240
2018-12-25@20:20 | Rev.... | 1 | ..4d | 2018-12-29 | 240  
..  
An edited text is a messy looking; do you mind if i delete it and make a new post ?

**Comment by dedu6ka** [comment]
My suggestion relates to Language cards.  
I would suspend the Card-1 'after' a certain number of  Good reps in the row - in Review status.  
 Tag the suspended cards "Buried" - because there might be other susp cards.  
The Last Good ( or Easy) rep will be a Scheduled rep; so we should not count it because we may fail it.  
But we still need to save it in the History table - to reflect it in the  Retention chart, the least.  
..  
The BuryLimit should be in the config file, a default BuryLimit = 5 looks good.  
..  
If a card is too easy.  we should have the option to suspend it at will. 
If a card is too hard, we need to 'postpone' the bury ( default number of extra reps in the config file ).  
..  
OVERDUE cards.  If i fail to recall an overdue card, i will reschedule it; it will be registered as the '0' type  
in the History table. This rep should not be counted at all, i think.  
User should know when a card is overdue ( there is an addon which flashes a screen "OVERDUE card").  
In addition to that, there is an addon which shows the Overdue amount
The addon description will tell the user what actions are available if a card is overdue. 
..  
PS.  
To ensure the high level of retention for the Graduation interval,  
we have to have a good learning-steps-sequence, one that ensures 95-98% success rate for the Grad ivl.  
Can you track it ? Even a Table in the Options would do;  or  
Modify the Glutanimate's "True retention by interval" addon. or  
In Anki graph, add another chart "Graduating ivl" ;  or  
In Anki graph, add chart "Step2 step3 Graduating ivl" chart would be the best.  
eof

**Comment by lovac42** [comment]
I will look over what you said again when I have time. Have you tried the addon? https://github.com/lovac42/TheRoommateSwitch

It already does most of these except for tagging and automation. So after swapping the sibling card, you have to manually suspend or bury the current card  or grade it, which will be logged. The sibling will then show up as the next review.

As for overdue and retention, I think I understand the gist of what you want from this. I have been using something similar, but it won't be shared as open source any time soon. So I'm not including those code in this project. It is a good idea though.

**Comment by dedu6ka** [comment]
Error message on loading:  
http://prntscr.com/mrlhs5

**Comment by lovac42** [comment]
Sorry I assumed you know how to install these.
Setup a zip: https://github.com/lovac42/TheRoommateSwitch/releases/tag/v0.1.1b

If you don't know how to unzip, I will ban you. Seriously.

**Comment by dedu6ka** [comment]
Go ahead, make your day   :-) 
 I don't do unzipping for living.
